src/index.js

```diff
@@ -7,9 +7,11 @@
 const { Stack } = require("./stack");
 const { Provider } = require("./provider");
 const { Backend } = require("./backend");
+const { plan } = require("./run");
 
 module.exports = {
   Stack,
   Provider,
-  Backend
+  Backend,
+  plan
```

Comment:
Should we really export internal functions like `plan`? I'm not sure if we really want to allow someone to use `plan` inside of a `stack.js`.

I think this is necessary to make the `terrastack-cli` work, right? But is there no other way? Maybe we can move all internal exports in `internal/index.js` and then use `require("terrastack/internal")` in the cli repo?

Reply:
That's still in progress and was just an intermediate step. I don't think it's a good idea to expose methods like this here. 

Was mostly trying to separate modules (the stack and its orchestration). The goal is, to be able somehow subscribe to state changes and logs eventually (compared to, doing this somewhere deep down in the code base)

And yes, the CLI plays a role for this goal. However, this is regardless of where the CLI lives.